src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileRepository.cs

```diff
@@ -29,5 +30,17 @@ public async Task<File> Get(Guid id)
             return await _contentDbContext.Files
                 .SingleAsync(f => f.Id == id);
         }
+
+        public async Task<string> GetSubjectName(Guid releaseId, Guid subjectId)
```

Comment:
The `FileRepository` is dedicated to files and there's a separate repository already for `ReleaseFiles`. There's even a `ReleaseDataFileRepository` for the things more specific to data files.

I would add a method to `ReleaseDataFileRepository` like `GetBySubjectId(releaseId, subjectId)` and return the ReleaseFile which is then more useful than returning string, and consume that.

Reply:
I saw this last night - because I noticed there were several FileRepository type things in ReleaseDataFilesService that could be removed and which could remove ReleaseDataFilesService's dependency on ContentDbContext, and at that point I noticed that this stuff should go in ReleaseFileRepository.

To remove ReleaseDataFileService's dependency on ContentDbContext - I'd need to also move GetAssociatedMetaFile, and create a new method which I called UnlinkReplacementFromOriginal. Worth doing?

Reply:
It sounds like you would be trying to shift more complex logic that belongs in the service into the repository which is typically for CRUD like database operations. Moving that kind of logic would likely make the repository less useful to other consumers especially if extra dependencies need to be added to it.